### ISSUE TITLE
`New shape and old shape must have the same number of elements.`, but the input/output shapes look ok

### ISSUE THREAD

**Issue by GuillaumeBadi** [tensorflow/tfjs#778]
Hello guys,
Not sure it is a bug, sorry if it's not

#### TensorFlow.js version
``` json
{
  "name": "tensortest",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "scripts": {
    "start": "TF_CPP_MIN_LOG_LEVEL=2 node example.js"
  },
  "dependencies": {
    "@tensorflow/tfjs": "^0.13.1",
    "@tensorflow/tfjs-node": "^0.1.18",
    "jpeg-js": "^0.3.4"
  }
}

```

#### Browser version
> Node.js

#### Describe the problem or feature request

I am using Tensorflow.js to start learning machine learning, and I experience an error while trying to use the fit method with a batched dataset (1 by 1 for an easy example to reproduce)

I have got a few images 600x600x3 (just one for the example) that I want to classify (one output node: 1 or 0)

#### Code to reproduce the bug / link to feature request
https://gist.github.com/GuillaumeBadi/21c3e1de42abe913132940a773769984

TDLR;
> A quick summary with the model and the training loop:
> BATCH_SIZE is 1 for the example, but any number output the exact same result
``` javascript
function createModel() {
  const model = tf.sequential()
  model.add(tf.layers.conv2d({
    inputShape: [600, 600, 3],
    kernelSize: 60,
    filters: 50,
    strides: 20,
    activation: 'relu',
    kernelInitializer: 'VarianceScaling'
  }))
  model.add(tf.layers.maxPooling2d({
    poolSize: [20, 20],
    strides: [20, 20]
  }))
  model.add(tf.layers.conv2d({
    kernelSize: 5,
    filters: 100,
    strides: 20,
    activation: 'relu',
    kernelInitializer: 'VarianceScaling'
  }))

  model.add(tf.layers.maxPooling2d({
    poolSize: [20, 20],
    strides: [20, 20]
  }))
  model.add(tf.layers.flatten())
  model.add(tf.layers.dense({
    units: 1,
    kernelInitializer: 'VarianceScaling',
    activation: 'softmax'
  }))
  return model
}

async function train() {
  try {
    const model = createModel()
    const optimizer = tf.train.sgd(LEARNING_RATE)

    model.compile({
      optimizer,
      loss: 'meanSquaredError',
      metrics: ['accuracy'],
    })

    const batches = await loadDataset()

    // For each batch in the dataset
    for (let i = 0; i < batches.length; i++) {

      const batch = batches[i]
      const ys = batch.ys

      console.log({
        xsShape: batch.xs.shape,
        ysShape: batch.ys.shape,
      })
      // { xsShape: [ 1, 600, 600, 3 ], ysShape: [ 1 ] }

      // ==== CODE THROWS HERE ====
      const history = await model.fit(
        batch.xs, ys,
        { batchSize: BATCH_SIZE, epochs: 1 },
      )
      // ==========================

      console.log('This will not print anything')

      const loss = history.history.loss[0]
      const accuracy = history.history.acc[0]

      console.log({ loss, accuracy })
    }
  } catch (e) {
    console.log(e)
  }
}

```

#### Stack trace

```
Error: new shape and old shape must have the same number of elements.
    at Object.assert (/Users/person/nn/node_modules/@tensorflow/tfjs-core/dist/util.js:36:15)
    at reshape_ (/Users/person/nn/node_modules/@tensorflow/tfjs-core/dist/ops/array_ops.js:271:10)
    at Object.reshape (/Users/person/nn/node_modules/@tensorflow/tfjs-core/dist/ops/operation.js:23:29)
    at Tensor.reshape (/Users/person/nn/node_modules/@tensorflow/tfjs-core/dist/tensor.js:273:26)
    at Object.derB [as $b] (/Users/person/nn/node_modules/@tensorflow/tfjs-core/dist/ops/binary_ops.js:32:24)
    at _loop_1 (/Users/person/nn/node_modules/@tensorflow/tfjs-core/dist/tape.js:90:47)
    at Object.backpropagateGradients (/Users/person/nn/node_modules/@tensorflow/tfjs-core/dist/tape.js:108:9)
    at /Users/person/nn/node_modules/@tensorflow/tfjs-core/dist/engine.js:334:20
    at /Users/person/nn/node_modules/@tensorflow/tfjs-core/dist/engine.js:91:22
    at Engine.scopedRun (/Users/person/nn/node_modules/@tensorflow/tfjs-core/dist/engine.js:101:23)
```

You can reproduce the error by setting an image path in the gist, install `jpeg-js` or use the provided `package.json` to decode the image, and `npm start`.

I thought the error was related to the `reshape` I used in the `loadDataset` method at first, but the error is only thrown during the `.fit` call and the stack trace mention a gradient propagation method, so I assume it's not related to the input or labels.

Is this error a bug, or do you think it is related to my code? Is there any right way to debug that? How can I understand this error message?

Anyway, thank you for the amazing work so far 👍

**Comment by GuillaumeBadi** [comment]
My bad, thanks a lot

**Comment by nkreeger** [comment]
Make sure you only include "@tensorflow/tfjs-node". The package now
includes "@tensorflow/tfjs".
